refactor(channels): remove NoCopy flag and optimize channel ops

sinkingsugar opened the pull request.

sinkingsugar committed:
refactor(channels): remove NoCopy flag and optimize channel ops

Remove NoCopy flag and related logic, replace boost::lockfree with
oneapi::tbb::concurrent_queue, and optimize channel operations for
better performance and safety. Update related tests accordingly.

sinkingsugar committed:
feat(channels): add buffer size to Consumers parameters

Extends Consumers struct to return buffer size as third parameter

sinkingsugar committed:
bump crdt-lite

guusw commented on shards/modules/channels/channels.hpp:
```diff
@@ -25,55 +24,61 @@ struct DummyChannel : public ChannelShared {
+    if (_data.try_pop(value)) {
+      func(std::forward<OwnedVar>(value));
+      if constexpr (Recycle) {
+        _recycle.push(std::move(value));
```
Maybe add a limit , cause if you only use the moving push it might grow the recycle list endlessly

sinkingsugar committed:
refactor(channels): Improve MPMCChannel push/pop operations

Introduce try_unrecycle and push_unsafe methods, modify push to use references, and reorder functions for better clarity and efficiency

sinkingsugar committed:
refactor(channels): use push_clone instead of push

Replace push with push_clone in MPMCChannel, Produce, and Broadcast
to ensure proper value cloning and prevent potential data races
